gemd/demo/cake.py

```diff
@@ -198,31 +233,37 @@ def make_cake_spec(tmpl=None):
     if tmpl is None:
         tmpl = make_cake_templates()
 
-    count = dict()
-
     def ingredient_kwargs(material):
         # Pulls the elements of a material that all ingredients consume out
-        count[material.name] = count.get(material.name, 0) + 1
         return {
-            "name": "{} input{}".format(material.name.replace('Abstract ', ''),
-                                        " (Again)" * (count[material.name] - 1)),
+            "name": material.name.lower(),
             "tags": list(material.tags),
             "material": material
         }
 
+    def material_kwargs(material_name, process_tmpl_name, process_kwargs):
+        # Handles some nicities in naming
+        return {
```

Comment:
This pattern probably deserves some documentation.  I see what its trying to do, but its not a pattern that I've seen before.

Reply:
Migrated to a wrapper for the constructor (the inverse of this, which I believe is a more common approach)